### ISSUE TITLE
spikes

### ISSUE THREAD

**Issue by notconvergingwtf** [deepinx/sdu-face-alignment#1]
Hi, do you have any suggestions on the next problem:
While training sdu(nadam,lr=0.00025), this is the loss on validation test:
![image](https://user-images.githubusercontent.com/48060406/53510422-fc2d1e00-3ace-11e9-9454-65b328d3ec94.png)
Different model on the same training data was fine
Also, while training, lossvalue=nan starts to appear

**Comment by deepinx** [comment]
I just set `network.sdu.net_coherent = True` and revise line 579  of sym_heatmap.py to `coherent_weight = 0.001`, it seems this nan problem can be solved.

**Comment by notconvergingwtf** [comment]
Okay, thanks
Sorry, but how did you manage to figure this out? Seems, that `network.sdu.net_coherent = True` stands for leaving only this image transformations, that doesn't affect heatmap? How does this affect accuracy?

**Comment by deepinx** [comment]
I did this following the guides of the origial paper, as in the paper  `Therefore, we employ the CE loss for Lp-g and the MSE loss for Lp-p, respectively. l is empirically set as 0:001 to guarantee convergence`

**Comment by notconvergingwtf** [comment]
Big thanks

**Comment by notconvergingwtf** [comment]
Hi, its me again. Well, after some train time, here what i got: 
![image](https://user-images.githubusercontent.com/48060406/54115864-83975d00-43fe-11e9-8daf-8d32a75938fc.png)
It doesnt look like overfitting, seems like some problems with convergence. Have you met the same problem?

**Comment by notconvergingwtf** [comment]
Hi, its me again. After some training time, here what i have:
![image](https://user-images.githubusercontent.com/48060406/54116004-e1c44000-43fe-11e9-8e75-2fa91a898d18.png)
It doesnt look like overfitting on train, may be some problems with convergence.. Have you met the same problem?

**Comment by deepinx** [comment]
What batch size and lr do you use?  You can try different batch size or lr, perhaps it can solve your problem.

**Comment by notconvergingwtf** [comment]
Batch size is 16. Well, as you can see, decreasing lr only delays time till spikes appear

**Comment by deepinx** [comment]
I am using batch-size 16 and lr 0.00002 at the first several epochs. The spike did not appear. You can try the following commands:
```
NETWORK='sdu'
MODELDIR='./model_2d'
mkdir -p "$MODELDIR"
PREFIX="$MODELDIR/$NETWORK"
LOGFILE="$MODELDIR/log_$NETWORK"

CUDA_VISIBLE_DEVICES='0' python -u train.py --network $NETWORK --prefix "$PREFIX" --per-batch-size 16 --lr 0.00002 --lr-step '16000,24000,30000' > "$LOGFILE" 2>&1 &
```
